video/out/vo_gpu_next.c

```diff
@@ -861,6 +861,40 @@ static void update_tm_viz(struct pl_color_map_params *params,
     params->visualize_hue = M_PI / 4.0;
 }
 
+static void rearrange_frame_mix(struct pl_frame_mix *mix) {
+    // Internal libplacebo frame mixing limit
+    // https://code.videolan.org/videolan/libplacebo/-/blob/426f346762180e6496e714a087be9561a9a1e236/src/renderer.c#L3254
+    static const int frame_mix_limit = 16;
+
+    if (mix->num_frames <= frame_mix_limit)
+        return;
+
+    int nearest = 0;
+    float best_dist = fabsf(mix->timestamps[nearest]);
+    for (int i = 1; i < mix->num_frames; i++) {
+        float dist = fabsf(mix->timestamps[i]);
+        if (dist >= best_dist)
+            break;
+        best_dist = dist;
+        nearest = i;
+    }
+
+    int count = frame_mix_limit / 2;
+    int start = nearest - count;
+    if (start < 0) {
+        count -= start;
+        start = 0;
+    }
+    int end = MPMIN(nearest + count, mix->num_frames);
+
+    for (int i = start, j = 0; i < end; i++, j++) {
+        ((struct pl_frame **)mix->frames)[j] = (struct pl_frame *)mix->frames[i];
+        ((uint64_t *)mix->signatures)[j] = mix->signatures[i];
+        ((float *)mix->timestamps)[j] = mix->timestamps[i];
+    }
+    mix->num_frames = end - start;
+}
```

Comment:
FWIW this sound like the kind of helper libplacebo should provide

Reply:
The presence of this helper function may not be necessary. Ideally, libplacebo should offer a mechanism (perhaps within pl_queue_params) to cap the number of frames in the mix. Alternatively, it should not extend the radius indefinitely but rather impose an internal limit, preferably one that is compatible with the renderer's capabilities.

Additionally, the renderer.c file should be designed to handle larger mixes without crashing. It could select the most significant frames for processing. Overmixing numerous frames is impractical as it leads to excessive processing times.

It can go to infinity with this line https://code.videolan.org/videolan/libplacebo/-/blob/426f346762180e6496e714a087be9561a9a1e236/src/utils/frame_queue.c#L814

Ideally, such issues should be resolved upstream. However, in the absence of fix #12873 (another upstream correction to pl_queue), the problem becomes more pronounced, particularly when adjusting playback speed. This is complicated by the inability to reliably disable interpolation due to libplacebo's internal VPS estimation. This means that even after resetting the speed, it might still use an outdated values unknown to mpv. (and still there is no api to limit frames, but then we could use pl_render_image and avoid mix)

An alternative solution could involve deactivating anti-aliasing for tscale. However, since libplacebo uses a single skip_anti_aliasing setting for all scales, turning it off for dscale too is not good, as it would result in undesirable aliasing effects.

Saying that I'm even fine NOT fixing it on mpv side and defer to next libplacebo release. But I feel like some workaround is better to avoid introducing this issue on users in the meantime.

Note that this issue is related to pretty recent libplacebo improvement https://code.videolan.org/videolan/libplacebo/-/commit/7872af19ccc88da39cf64ddc9beb3df68e888d89 but unfortunate it is included in the latest release.